File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        seen = {}

        for i in range(len(numbers)):
            num = numbers[i]
            need = target - num
            if need in seen:
                return [seen[need], i + 1]
            seen[num] = i + 1

File: Data Structures & Algorithms/is-anagram/submission-0.js
class Solution {
    /**
     * @param {string} s
     * @param {string} t
     * @return {boolean}
     */
    isAnagram(s, t) {
        const seen = {}
        
        for(const c of s){
            if(!seen[c]){
                seen[c] = 1
            } else {
                seen[c] += 1
            }
        }

        for(const c of t){
            if(!seen[c]){
                return false
            }

            seen[c] -= 1
        }

        return Object.values(seen).filter((v) => v !== 0).length !== 0 ? false : true
    }
}
